game-core/src/main/java/games/strategy/engine/data/export/GameDataExporter.java

```diff
@@ -267,409 +217,348 @@ private void relationshipInitialize(final GameData data) {
+  private Initialize.ResourceInitialize resourceInitialize(final GameData data) {
+    final List<Initialize.ResourceInitialize.ResourceGiven> resourcesGiven = new ArrayList<>();
+
     for (final GamePlayer player : data.getPlayerList()) {
       for (final Resource resource : data.getResourceList().getResources()) {
         if (player.getResources().getQuantity(resource.getName()) > 0) {
-          xmlfile
-              .append("            <resourceGiven player=\"")
-              .append(player.getName())
-              .append("\" resource=\"")
-              .append(resource.getName())
-              .append("\" quantity=\"")
-              .append(player.getResources().getQuantity(resource.getName()))
-              .append("\"/>\n");
+          resourcesGiven.add(
+              Initialize.ResourceInitialize.ResourceGiven.builder()
+                  .player(player.getName())
+                  .resource(resource.getName())
+                  .quantity(player.getResources().getQuantity(resource.getName()))
+                  .build());
         }
       }
     }
-    xmlfile.append("        </resourceInitialize>\n");
+    return Initialize.ResourceInitialize.builder().resourcesGiven(resourcesGiven).build();
   }
 
-  private void unitInitialize(final GameData data) {
-    xmlfile.append("        <unitInitialize>\n");
+  private Initialize.UnitInitialize unitInitialize(final GameData data) {
+    final List<Initialize.UnitInitialize.UnitPlacement> unitPlacements = new ArrayList<>();
+
     for (final Territory terr : data.getMap().getTerritories()) {
       final UnitCollection uc = terr.getUnitCollection();
       for (final GamePlayer player : uc.getPlayersWithUnits()) {
         final IntegerMap<UnitType> ucp = uc.getUnitsByType(player);
         for (final UnitType unit : ucp.keySet()) {
-          if (player == null || player.getName().equals(Constants.PLAYER_NAME_NEUTRAL)) {
-            xmlfile
-                .append("            <unitPlacement unitType=\"")
-                .append(unit.getName())
-                .append("\" territory=\"")
-                .append(terr.getName())
-                .append("\" quantity=\"")
-                .append(ucp.getInt(unit))
-                .append("\"/>\n");
-          } else {
-            xmlfile
-                .append("            <unitPlacement unitType=\"")
-                .append(unit.getName())
-                .append("\" territory=\"")
-                .append(terr.getName())
-                .append("\" quantity=\"")
-                .append(ucp.getInt(unit))
-                .append("\" owner=\"")
-                .append(player.getName())
-                .append("\"/>\n");
-          }
+
+          unitPlacements.add(
+              Initialize.UnitInitialize.UnitPlacement.builder()
+                  .owner(
+                      player == null || player.getName().equals(Constants.PLAYER_NAME_NEUTRAL)
+                          ? null
```

Comment:
Codacy found an issue: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6193173)